R/derive_merged.R

```diff
@@ -986,20 +986,183 @@ get_not_mapped <- function() {
 #'   "2",         "WEEK 1",  "INV-N1"
 #' )
 #'
-#' derive_var_merged_summary(
+#' derive_vars_merged_summary(
 #'   adsl,
 #'   dataset_add = adtr,
 #'   by_vars = exprs(USUBJID),
 #'   filter_add = AVISIT == "BASELINE",
 #'   new_vars = exprs(LESIONSBL = paste(LESIONID, collapse = ", "))
 #' )
 #'
+derive_vars_merged_summary <- function(dataset,
+                                       dataset_add,
+                                       by_vars,
+                                       new_vars = NULL,
+                                       filter_add = NULL,
+                                       missing_values = NULL) {
+  assert_vars(by_vars)
+  by_vars_left <- replace_values_by_names(by_vars)
+  by_vars_right <- chr2vars(paste(vars2chr(by_vars)))
+  # once new_var is removed new_vars should be mandatory
+  assert_expr_list(new_vars, named = TRUE, optional = TRUE)
+  filter_add <-
+    assert_filter_cond(enexpr(filter_add), optional = TRUE)
+  assert_data_frame(
+    dataset,
+    required_vars = by_vars_left
+  )
+  assert_data_frame(
+    dataset_add,
+    required_vars = expr_c(by_vars_right, extract_vars(new_vars))
+  )
+
+  # Summarise the analysis value and merge to the original dataset
+  # If for one of the new variables no summary function is used, i.e., more than
+  # one record is created per by group, the error from signal_duplicates_records()
+  # need to be updated and the warning from dplyr needs to be suppressed as it
+  # is misleading.
+  tryCatch(
+    withCallingHandlers(
+      derive_vars_merged(
+        dataset,
+        dataset_add = derive_summary_records(
+          dataset_add = dataset_add,
+          by_vars = by_vars_right,
+          filter_add = !!filter_add,
+          set_values_to = new_vars,
+        ) %>%
+          select(!!!by_vars_right, names(new_vars)),
+        by_vars = by_vars,
+        missing_values = missing_values
+      ),
+      warning = function(cnd) {
+        if (any(str_detect(
+          cnd$message,
+          fixed("Returning more (or less) than 1 row per `summarise()` group was deprecated")
+        ))) {
+          cnd_muffle(cnd)
+        }
+      }
+    ),
+    duplicate_records = function(cnd) {
+      cli_abort(
+        c(
+          paste(
+            "After summarising, the dataset contains duplicate records with",
+            "respect to {.var {cnd$by_vars}}."
+          ),
+          paste(
+            "Please check {.arg new_vars} if summary functions like {.fun mean},",
+            "{.fun sum}, ... are used on the right hand side."
+          ),
+          i = "Run {.run admiral::get_duplicates_dataset()} to access the duplicate records"
+        ),
+        class = cnd$class,
+        call = NULL,
+        by_vars = cnd$by_vars
+      )
+    }
+  )
+}
+
+#' Merge Summary Variables
+#'
+#' @description Merge a summary variable from a dataset to the input dataset.
```

Comment:
The deprecation badge and text should be added here.

Reply:
badge added - great catch btw i hadn't forgotten about those!!